catalog/rocks/cataloger.go

```diff
@@ -378,8 +566,51 @@ func (c *cataloger) GetCommit(ctx context.Context, repository string, reference
 	return catalogCommitLog, nil
 }
 
-func (c *cataloger) ListCommits(ctx context.Context, repository string, branch string, fromReference string, limit int) ([]*catalog.CommitLog, bool, error) {
-	panic("not implemented") // TODO: Implement
+func (c *cataloger) ListCommits(ctx context.Context, repository string, _ string, fromReference string, limit int) ([]*catalog.CommitLog, bool, error) {
+	repositoryID, err := graveler.NewRepositoryID(repository)
+	if err != nil {
```

Comment:
We should have here two kind of "froms", one is the Reference the Log is built out of and the second is for the pagination.

Reply:
resolved the branch and iterated until from ref if passed any value